process-engine/process_engine_core
# ✨ Add a State Flag to the Correlation Data Model

**Requires:**
https://github.com/process-engine/process_engine_contracts/pull/102

**Changes:**

1. Add the `state` and `error` properties to correlations.
2. Simplify the state-specific correlation queries used by the `CorrelationService`.
    - Use the correlation's state flag to perform state related queries.
    - Remove a lot of code clutter, caused by the previously required usage of the FlowNodeInstance repository.

**Issues:**

Related: https://github.com/process-engine/process_engine_runtime/issues/229

PR: #238 

## How can others test the changes?

Execute some ProcessInstances and see that the state of the corresponding correlation gets updated accordingly to `running|finished|error`.

## PR-Checklist

Please check the boxes in this list after submitting your PR:
